src/streamlink_cli/argparser.py

```diff
@@ -467,6 +467,15 @@ def boolean(value):
 
 stream = parser.add_argument_group("Stream options")
 stream.add_argument(
+    "--url",
+    dest="url_param",
+    metavar="URL",
+    help="""
+    Alternative parameter to specify a URL to attempt to
```

Comment:
We could suppress the help and then explain it as a special case in the docs.
I'll have a go at writing something a bit more friendly though :-)